tests/test_preprocessing_utils.py

```diff
@@ -180,9 +187,417 @@ def test_toy_dataset(self):
+        )
+        url_processor = UrlPreprocessor(col_to_store_metadata=col_to_store_metadata_url, col_url="url")
+        timestamp_processor = TimestampPreprocessor(
+            col_to_store_metadata=col_to_store_metadata_timestamp, col_metadata_url=col_to_store_metadata_url
+        )
+        website_processor = WebsiteDescPreprocessor(
+            col_to_store_metadata=col_to_store_metadata_website_desc, col_metadata_url=col_to_store_metadata_url
+        )
+        entity_processor = EntityPreprocessor(
+            base_url="", path_wiki_db="", col_to_store_metadata=col_to_store_metadata_entities
+        )
+
+        features = Features(
+            {
+                "doc_html": Value("string"),
+                "metadata": [
+                    {
+                        "char_end_idx": Value("int64"),
+                        "char_start_idx": Value("int64"),
+                        "html_attrs": {"attrs": [Value("string")], "values": [Value("string")]},
+                        "key": Value("string"),
+                        "relative_end_pos": Value("int64"),
+                        "relative_start_pos": Value("int64"),
+                        "type": Value("string"),
+                        "value": Value("string"),
+                        "ent_desc": Value("string"),
+                    }
+                ],
+                "text": Value("string"),
+                "url": Value("string"),
+            }
+        )
+
+        # Apply function
+        ds = Dataset.from_dict(self.init_dict)
+        ds = ds.map(lambda ex: html_processor.preprocess(ex), batched=True, batch_size=3, features=features)
+        ds = ds.map(lambda ex: url_processor.preprocess(ex), batched=True, batch_size=3, features=features)
+        ds = ds.map(lambda ex: timestamp_processor.preprocess(ex), batched=True, batch_size=3, features=features)
+        ds = ds.map(lambda ex: website_processor.preprocess(ex), batched=True, batch_size=3, features=features)
+        ds = ds.map(lambda ex: entity_processor.preprocess(ex), batched=True, batch_size=3, features=features)
+
+        self.assertEqual(ds[:][col_to_store_text], self.target_texts)
+
+        for id, metadata_example in enumerate(self.target_metadata_html):
+            for metadata in metadata_example:
+                metadata.update(
+                    {
+                        "ent_desc": None,
+                    }
+                )
+                self.assertIn(metadata, ds[id][col_to_store_metadata_html])
+
+        for id, metadata_example in enumerate(self.target_metadata_url):
+            for metadata in metadata_example:
+                metadata.update(
+                    {
+                        "char_end_idx": None,
+                        "char_start_idx": None,
+                        "relative_end_pos": None,
+                        "relative_start_pos": None,
+                        "html_attrs": None,
+                        "ent_desc": None,
```

Comment:
Note here that with the datasets library all dictionaries in the lists of a unique column must have the same keys!